src/main/scala/org/broadinstitute/dsde/workbench/sam/openam/PostgresAccessPolicyDAO.scala

```diff
@@ -59,10 +59,15 @@ class PostgresAccessPolicyDAO(protected val dbRef: DbReference,
     }
 
     if (roleActionValues.nonEmpty) {
+      val ra = RoleActionTable.syntax("ra")
+
+      samsql"""delete from ${RoleActionTable as ra}
+              where ${ra.resourceRoleId} in (${resourceTypeRoles.map(_.id)})"""
+        .update().apply()
+
```

Comment:
I think I prefer leaving the `on conflict do nothing` below and having the delete statement above only delete actions that are not in the list (hint, use a `not in` clause). That way the usual behavior on startup is effectively not database changes.

I would follow the same pattern in `overwriteActionPatterns`. I think the leaving `overwriteRoles` the way you currently have it is ok. Something about removing rows and re-adding them on every restart of sam makes me shiver and in these cases it is pretty easy not to do so.

It is probably also easy to deprecate only what is needed in `overwriteRoles` too and be sure to undeprecate what was re-added but I have less strong emotions about that.